File: predator-tck/src/main/java/io/micronaut/data/tck/entities/Country.java
package io.micronaut.data.tck.entities;

import io.micronaut.data.annotation.*;

import java.util.Set;
import java.util.UUID;

@MappedEntity
public class Country {

    @Id
    @AutoPopulated
    private UUID uuid;
    private String name;

    @Relation(value = Relation.Kind.ONE_TO_MANY, mappedBy = "country")
    private Set<CountryRegion> regions;

    public Country(String name) {
        this.name = name;
    }

    public String getName() {
        return name;
    }

    public UUID getUuid() {
        return uuid;
    }

    public void setUuid(UUID uuid) {
        this.uuid = uuid;
    }

    public Set<CountryRegion> getRegions() {
        return regions;
    }

    public void setRegions(Set<CountryRegion> regions) {
        this.regions = regions;
    }
}

File: doc-examples/jdbc-example-python/src/main/python/example/Country.py
from dataclasses import dataclass, field
from typing import TYPE_CHECKING, Annotated
from uuid import UUID

from micronaut.data.annotation import AutoPopulated, Id, MappedEntity, Relation

if TYPE_CHECKING:
    from example.CountryRegion import CountryRegion


# tag::country[]
@MappedEntity
@dataclass
class Country:
    name: str
    regions: Annotated["set[CountryRegion]", Relation(value="ONE_TO_MANY", mappedBy="country")] = field(default_factory=set)
    uuid: Annotated[UUID | None, Id, AutoPopulated] = None
# end::country[]
